### PR TITLE
added SWMM5 engine selector

### PR DESCRIPTION
**Pull request by bemcdonnell**
Covers Issue #77 

So put your DLL in the lib folder and run the following:

```python
>>> import pyswmm
>>> pyswmm.lib.use("YourDllHere") # <-- Not a path 
>>>
>>> from pyswmm import Simulation
...
```

### CODE REVIEW

**Comment by bemcdonnell** [comment]
Also Covers #74

**Comment by goanpeca** [comment]
I think that this is not really a good solution in general :-\

The real need here is that people need to have the correct vs runtime. This is out of the scope for a python package, so this should be fixed by telling people on the readme to install the correct runtime for the specific version.

**Comment by bemcdonnell** [comment]
@goanpeca, I'm not sure what you mean by correct vs runtime. Could you elaborate?

This is sort of like matplotlib giving you the option to select a backend.  My company has a version of swmm that has expanded capability beyond standard EPASWMM. I want the option to use either the SWMM that get distributed with pyswmm or my company's version of swmm.

**Comment by bemcdonnell** [comment]
Maybe I didn't describe what this is doing well enough. 

This let's you choose what swmm dll you want to us... Maybe you interpreted this as adding runtime libs to support the swmm.dll? Nope :-)

If they users has a special version of swmm, they drop their version in the lib/windows folder and then call the 

'pyswmm.lib.use("swmm5-extended")'

**Comment by bemcdonnell** [comment]
@goanpeca this feature has lots of uses too. We can use pyswmm to run regression tests to compare old swmm engine output to new. The ability to choose an engine was an original request of @michaeltryby.

**Comment by dickinsonre** [comment]
@bemcdonnell One problem with using older engines of SWMM5 is that in some but not many cases the data on the line that has to be parsed is different.  I do have good success with newer SWMM5 models in older versions but the SWMM5 inp file usually has to pass through the older version GUI to iron out inp problems.

**Comment by bemcdonnell** [comment]
@dickinsonre, you certainly raise a good point.  In your experience, how many versions back to SWMM5.0 would you say there has been consistency in the structure? Assuming we can get the correct structure in place for each respective model, unless the inflow equations change fundamentally, we should be able to understand and quantify the changes in engine behavior.  At the same time, that might be a "risky" exercise. This discussion might be better suited for the SWMM repo

**Comment by dickinsonre** [comment]
I just tried a few of the older versions with a new inp file and you are okay going back to 5.0.020 at least but  older version of SWMM5 had problems with the INFIL section and other sections and did run.  The pascal uexport.pas code is available for all all versions so a universal exporter may be possible (if desired).

**Comment by goanpeca** [comment]
@bemcdonnell I still don't understand how

>If the user has a special version of swmm, they drop their version in the lib/windows folder and then call the
'pyswmm.lib.use("swmm5-extended")'


If it is special and adds extra methods, well then the whole pyswmm needs to account for these extra stuff. For older versions, its similar but I can understand this use case better.

**Comment by bemcdonnell** [comment]
@goanpeca, pyswmm is independent. No features will be added to pyswmm that will not work with the "common" distributed version of swmm with pyswmm. We have compiled the toolkit API into previous versions of swmm and want to take advantage of that. I see lots of value to this feature. This is also for the "power users". 

The discussion is. It so much on "does this add value" but more so does the implementation fit with the project. Having moved most of the Mechanics to pyswmm.lib.__init__.py, I have enabled us to be able to plan for future OS/ARCH and Thererin allows you to use your version of swmm. 

One other great point. If you want to run your project in an older version of EPASWMM that does not have the toolkit API, the basic functionally will still work.  "swmm_open" "swmm_run" have been standard entry points since swmm 5.0. 

My implementation here provides a much cleaner way to select which engine you want to use. All the paths are handled automagicaly as long as your version of swmm is sitting next to the binaries that come with pyswmm.

**Comment by goanpeca** [comment]
Sure go ahead, was just asking to understand better ;-)

File: pyswmm/lib/__init__.py [hunk 1]
```diff
@@ -10,7 +10,66 @@
...
+
+
+
+# Local Path
 HERE = os.path.abspath(os.path.dirname(__file__))
 
+
+# Platform Detection
+def _platform():
+    """ Folder based on platform """
```
Comment (goanpeca) [inline pyswmm/lib/__init__.py]: `"""Folder based on platform."""`

File: pyswmm/lib/__init__.py [hunk 2]
```diff
@@ -10,7 +10,66 @@
...
-LIB_SWMM_WIN_32 = os.path.join(HERE, 'windows', 'swmm5.dll').replace('\\', '/')
+if os.name == 'nt':
+    LIB_SWMM = os.path.join(HERE, _platform(),
+                            'swmm5.dll').replace('\\', '/')
+
+
+class _DllPath(object):
+    """
+    DllPath Object.
+    """
```
Comment (goanpeca) [inline pyswmm/lib/__init__.py]: `"""DllPath Object."""`

File: pyswmm/lib/__init__.py [hunk 3]
```diff
@@ -10,7 +10,66 @@
...
+    def __init__(self):
+        self._dll_loc = LIB_SWMM
+    @property
+    def dll_loc(self):
+        """Get/set DLL Name """
+        return self._dll_loc
+    @dll_loc.setter
```
Comment (goanpeca) [inline pyswmm/lib/__init__.py]: Add a space here

File: pyswmm/lib/__init__.py [hunk 4]
```diff
@@ -10,7 +10,66 @@
...
+                            'swmm5.dll').replace('\\', '/')
+
+
+class _DllPath(object):
+    """
+    DllPath Object.
+    """
+    def __init__(self):
+        self._dll_loc = LIB_SWMM
+    @property
```
Comment (goanpeca) [inline pyswmm/lib/__init__.py]: Add a space here

File: pyswmm/lib/__init__.py [hunk 5]
```diff
@@ -10,7 +10,66 @@
...
+    def dll_loc(self):
+        """Get/set DLL Name """
```
Comment (goanpeca) [inline pyswmm/lib/__init__.py]: `"""Get/set DLL Name."""`

File: pyswmm/lib/__init__.py [hunk 6]
```diff
@@ -10,7 +10,66 @@
...
+        return self._dll_loc
+    @dll_loc.setter
+    def dll_loc(self, value):
+        """Set DLL Name """
```
Comment (goanpeca) [inline pyswmm/lib/__init__.py]: """Set DLL Name."""

File: pyswmm/lib/__init__.py [hunk 7]
```diff
@@ -10,7 +10,66 @@
...
+        self._dll_loc = value
+    def __call__(self):
+        """ """
```
Comment (goanpeca) [inline pyswmm/lib/__init__.py]: ???

File: pyswmm/lib/__init__.py [hunk 8]
```diff
@@ -10,7 +10,66 @@
...
+        return self._dll_loc
+
+
+# Initialize dll path object
+DLL_SELECTION = _DllPath()
+
+
+def use(arg):
+    """
+    Set the SWMM5 DLL
```
Comment (goanpeca) [inline pyswmm/lib/__init__.py]: `Set the SWMM5 DLL.`  (the period!)

**Pull request merged**